File: src/Services/JukeBox/JukeBox.Core/Models/Player.cs
namespace JukeBox.Core.Models;

public class PlayerPlayResumeRequest
{
    public string context_uri { get; set; }
    public int position_ms { get; set; }
    public OffSet offset { get; set; }
}

public class OffSet
{
    public int position { get; set; }
}

public class Actions
{
    public bool interrupting_playback { get; set; }
    public bool pausing { get; set; }
    public bool resuming { get; set; }
    public bool seeking { get; set; }
    public bool skipping_next { get; set; }
    public bool skipping_prev { get; set; }
    public bool toggling_repeat_context { get; set; }
    public bool toggling_shuffle { get; set; }
    public bool toggling_repeat_track { get; set; }
    public bool transferring_playback { get; set; }
}

public class Context
{
    public string type { get; set; }
    public string href { get; set; }
    public ExternalUrls external_urls { get; set; }
    public string uri { get; set; }
}

public class Device
{
    public string id { get; set; }
    public bool is_active { get; set; }
    public bool is_private_session { get; set; }
    public bool is_restricted { get; set; }
    public string name { get; set; }
    public string type { get; set; }
    public int volume_percent { get; set; }
}

public class PlayerStateResponse
{
    public Device device { get; set; }
    public string repeat_state { get; set; }
    public bool shuffle_state { get; set; }
    public Context context { get; set; }
    public int timestamp { get; set; }
    public int progress_ms { get; set; }
    public bool is_playing { get; set; }
    public Item item { get; set; }
    public string currently_playing_type { get; set; }
    public Actions actions { get; set; }
}

File: src/Web/spocifyapp/src/app/core/models/spocify/player.ts
import { IExternalUrls, IItem } from "./search";

export interface IPlayerPlayResumeRequest {
    context_uri: string;
    position_ms: number;
    offset: IOffSet;
}

export interface IOffSet {
    position: number;
}

export interface IActions {
    interrupting_playback: boolean;
    pausing: boolean;
    resuming: boolean;
    seeking: boolean;
    skipping_next: boolean;
    skipping_prev: boolean;
    toggling_repeat_context: boolean;
    toggling_shuffle: boolean;
    toggling_repeat_track: boolean;
    transferring_playback: boolean;
}

export interface IContext {
    type: string;
    href: string;
    external_urls: IExternalUrls;
    uri: string;
}

export interface IDevice {
    id: string;
    is_active: boolean;
    is_private_session: boolean;
    is_restricted: boolean;
    name: string;
    type: string;
    volume_percent: number;
}

export interface IPlayerStateResponse {
    device: IDevice;
    repeat_state: string;
    shuffle_state: boolean;
    context: IContext;
    timestamp: number;
    progress_ms: number;
    is_playing: boolean;
    item: IItem;
    currently_playing_type: string;
    actions: IActions;
}
